port_ocean/core/handlers/webhook/processor_manager.py

```diff
@@ -0,0 +1,209 @@
+from typing import Dict, Type, Set, Callable
+from fastapi import APIRouter, Request
+from loguru import logger
+import asyncio
+from dataclasses import dataclass
+
+from .webhook_event import WebhookEvent, WebhookEventTimestamp
+
+
+from .abstract_webhook_processor import AbstractWebhookProcessor
+from port_ocean.utils.signal import SignalHandler
+from port_ocean.core.handlers.queue import AbstractQueue, LocalQueue
+
+
+@dataclass
+class ProcessorRegistration:
+    """Represents a registered processor with its filter."""
+
+    processor: Type[AbstractWebhookProcessor]
+    filter: Callable[[WebhookEvent], bool]
+
+
+class WebhookProcessorManager:
+    """Manages webhook processors and their routes."""
+
+    def __init__(
+        self,
+        router: APIRouter,
+        signal_handler: SignalHandler,
+        max_event_processing_seconds: float = 90.0,
+        max_wait_seconds_before_shutdown: float = 5.0,
```

Comment:
lets add todo to have it as part of the configuration which could be later managed dynamically, for now until we don't have real usage there is no need to implement it now